src/test/ui/parser/if-attrs/deny-if-attr-validation.rs

```diff
@@ -0,0 +1,9 @@
+fn main() {
+    #[allow(warnings)] if true {} //~ ERROR attributes are not yet allowed on `if` expressions
```

Comment:
```rust
#[cfg(not(foo))]
if cond1 {
} else #[cfg(not(bar))] if cond2 {
} else #[cfg(not(baz))] {
}
```

doesn't parse - I have somthing similar in `src/test/ui/parser/if-attrs/else-attrs.rs`

Reply:
Right, hmm... the check-pass test does exercise the config behavior, but for extra robustness, could we add something like this?
```rust
// run-pass

fn main() {
    #[cfg(FALSE)]
    if true { panic!() }
}
```

Reply:
I'm confused - the whole point of this PR is if-attrs are allowed syntactically, but denied semantically. Your example will error with "attributes are not yet allowed on `if` expressions" both before and after this PR - it's just that the error is being emitted from a different place now.

Reply:
Ah right; I was thinking that cfg expansion might strip out the attribute. Can you add `#[cfg(FALSE)]` and `#[cfg_attr(FALSE, whatever)]` to this file to exercise that behavior for conditional compilation specifically?

Reply:
@Centril: Actually, you're right - that code does compile with this PR, though I think it probably shouldn't. Should I modify cfg-expansion to deny `#[cfg]` attributes on `if` expressions?

Reply:
That does seem like the conservative choice, although I have no idea how complicated it would be. Remember to also cover `cfg_attr`. But this is something that @petrochenkov should r+ :)